src/controller/quota/controller.go

```diff
@@ -19,23 +19,21 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/goharbor/harbor/src/lib"
 	"github.com/goharbor/harbor/src/lib/errors"
 	"github.com/goharbor/harbor/src/lib/log"
 	"github.com/goharbor/harbor/src/lib/orm"
 	"github.com/goharbor/harbor/src/lib/q"
-	redislib "github.com/goharbor/harbor/src/lib/redis"
 	"github.com/goharbor/harbor/src/pkg/quota"
 	"github.com/goharbor/harbor/src/pkg/quota/driver"
 	"github.com/goharbor/harbor/src/pkg/quota/types"
-	"github.com/gomodule/redigo/redis"
 
 	// quota driver
 	_ "github.com/goharbor/harbor/src/controller/quota/driver"
 )
 
 var (
-	// expire reserved resources when no actions on the key of the reserved resources in redis during 1 hour
-	defaultReservedExpiration = time.Hour
+	defaultRetryTimeout = time.Minute * 5
```

Comment:
It may cause some APIs failed by http time out, but the quota still be updated in the backend?

Reply:
Using the `select for update` has the same problem.

I think this is not a problem if the Nginx closed the TCP conn when the timeout happened. In this case, writing the data to the response writer will fail, and the tx will be abort.